package.json

```diff
@@ -278,7 +278,8 @@
     "load-grunt-tasks": "^3.4.0",
     "minimist": "^1.2.0",
     "mocha": "^2.4.5",
-    "request": "^2.81.0"
+    "request": "^2.81.0",
+    "sinon": "^5.0.3"
```

Comment:
Using sinon spies to verify the events are only emitted once. Let me know if you prefer something else.

Reply:
sinon is good and agree to use it
`v2.0.0` will drop to any version < 4. So I will temporarily comment out  the multiple call check and add it back in `v2.0.0`